File: js/logger/logger.js
class Logger {
  constructor(level = 'info') {
    this.levels = { debug: 0, info: 1, warn: 2, error: 3 };
    this.level = this.levels[level] !== undefined ? this.levels[level] : 1;
  }

  log(level, message, meta = {}) {
    if (this.levels[level] >= this.level) {
      console.log(JSON.stringify({
        timestamp: new Date().toISOString(),
        level: level.toUpperCase(),
        message,
        ...meta
      }));
    }
  }

  debug(msg, meta) { this.log('debug', msg, meta); }
  info(msg, meta) { this.log('info', msg, meta); }
  warn(msg, meta) { this.log('warn', msg, meta); }
  error(msg, meta) { this.log('error', msg, meta); }
}

module.exports = Logger;

File: go/logger/logger.go
package logger

import (
	"log/slog"
	"os"
)

func NewLogger(level string) *slog.Logger {
	var l slog.Level
	switch level {
	case "debug":
		l = slog.LevelDebug
	case "info":
		l = slog.LevelInfo
	case "warn":
		l = slog.LevelWarn
	case "error":
		l = slog.LevelError
	default:
		l = slog.LevelInfo
	}

	opts := &slog.HandlerOptions{Level: l}
	handler := slog.NewJSONHandler(os.Stdout, opts)
	return slog.New(handler)
}
